stl/inc/istream

```diff
@@ -206,6 +206,12 @@ private:
         return *this;
     }
 
+    void _Increment_gcount() {

```

Comment:
Like much of iostreams, this is separately compiled code even though it appears in `stl/inc`. (One indication is the presence of `__CLR_OR_THIS_CALL` calling conventions everywhere, although the hint is unrecognizable without already knowing about it.) So, adding this ordinary member function will affect the STL's export surface, meaning that this is an ABI break. :scream_cat:

(We have an internal build step that verifies the export surface but this not yet been ported to GitHub/CMake. Apologies!)

We can verify whether this affects the export surface by using `dumpbin /exports`:

```
S:\GitHub\STL\out\build\x86\out\bin\i386>dumpbin /exports msvcp140_oss.dll | findstr _Increment_gcount
        500  1F3 0000C5E0 ?_Increment_gcount@?$basic_istream@DU?$char_traits@D@std@@@std@@AAEXXZ = ?_Increment_gcount@?$basic_istream@DU?$char_traits@D@std@@@std@@AAEXXZ (private: void __thiscall std::basic_istream<char,struct std::char_traits<char> >::_Increment_gcount(void))
        501  1F4 0000C5E0 ?_Increment_gcount@?$basic_istream@GU?$char_traits@G@std@@@std@@AAEXXZ = ?_Increment_gcount@?$basic_istream@DU?$char_traits@D@std@@@std@@AAEXXZ (private: void __thiscall std::basic_istream<char,struct std::char_traits<char> >::_Increment_gcount(void))
        502  1F5 0000C5E0 ?_Increment_gcount@?$basic_istream@_WU?$char_traits@_W@std@@@std@@AAEXXZ = ?_Increment_gcount@?$basic_istream@DU?$char_traits@D@std@@@std@@AAEXXZ (private: void __thiscall std::basic_istream<char,struct std::char_traits<char> >::_Increment_gcount(void))
```

This indicates that your new member function is appearing in the export table, when instead we want `findstr` to find nothing here.

Possible workarounds:
* Don't add a member function at all, and instead repeat the `if`-statement.
* Extract this to be a non-member function that operates on `_Chcount` by reference. Only member functions (non-static and static) will be exported.
* Make this member function a template, like `_Common_extract_with_num_get` immediately above. This prevents it from being exported (the compiler can't instantiate a template without knowing what the template argument will be). I'm 99% sure you could say `template <class = void>` or `template <int = 0>` and that would be sufficient.

Reply:
`template <class = void>` helps. Thank you for explanation